app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt

```diff
@@ -110,9 +110,17 @@ class MenuDrawerAdapter @Inject constructor() : ListAdapter<Any, MenuDrawerViewH
             hasCollapsableDefaultFolder = temporaryHasCollapsableDefaultFolder
             hasCollapsableCustomFolder = temporaryHasCollapsableCustomFolder
 
-            // Footer
+            // Actions
             add(ItemType.DIVIDER)
-            add(MenuDrawerFooter(permissions, quotas))
+            add(ItemType.ACTIONS_HEADER)
+            if (areActionsExpanded) {
+                add(syncAutoConfigAction)
+                add(importMailsAction)
+                if (permissions?.canRestoreEmails == true) add(restoreMailsAction)
+            }
+
+            // Footer
+            add(MenuDrawerFooter(quotas))
```

Comment:
Instead of endlessly adding new items in this `formatList()` method, shouldn't we listen to sonar and actually reduce cognitive complexity in here? Each comment that separates an area could be its own method or something. Also, all these methods only used to create this very list could probably go into their own file so they are neatly kept together I guess